### PR TITLE
Upgrade to pyxform 1.5.0

### PR DESCRIPTION
**Pull request by duvld**
Upgrade `pyxform` dependency to latest version and fix broken tests.

Last of broken tests:
```
================================================== short test summary info ===================================================
FAILED onadata/apps/logger/tests/test_simple_submission.py::TestSimpleSubmission::test_corrupted_submission
FAILED onadata/apps/logger/tests/test_simple_submission.py::TestSimpleSubmission::test_simple_yes_submission
FAILED onadata/apps/logger/tests/test_simple_submission.py::TestSimpleSubmission::test_start_time_boolean_properly_set
FAILED onadata/apps/logger/tests/test_simple_submission.py::TestSimpleSubmission::test_start_time_submissions
FAILED onadata/apps/main/tests/test_csv_export.py::TestExport::test_csv_export_output
FAILED onadata/apps/main/tests/test_past_bugs.py::TestInputs::test_uniqueness_of_group_names_enforced
FAILED onadata/apps/main/tests/test_process.py::TestProcess::test_xls_export_content
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_build_sections_from_survey
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_child_record_parent_table_is_updated_when_sheet_is_renamed
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_delimiter_replacement_works_for_generated_gps_fields
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_delimiter_replacement_works_generated_multi_select_fields
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_generation_of_gps_fields_works
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_generation_of_mongo_encoded_fields_works
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_generation_of_multi_selects_works
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_to_xls_export_generates_valid_sheet_names
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_to_xlsx_export_respects_custom_field_delimiter
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_to_xlsx_export_works
FAILED onadata/apps/viewer/tests/test_exports.py::TestExports::test_column_header_delimiter_export_option
FAILED onadata/apps/viewer/tests/test_exports.py::TestExports::test_csv_http_response
FAILED onadata/apps/viewer/tests/test_exports.py::TestExports::test_csv_without_na_values
FAILED onadata/apps/viewer/tests/test_exports.py::TestExports::test_split_select_multiple_export_option
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_csv_column_indices_in_groups_within_repeats
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_csv_columns_for_gps_within_groups
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_csv_dataframe_export_to
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_csv_export_with_df_size_limit
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_format_mongo_data_for_csv
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_split_select_multiples
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_xls_columns
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_xls_columns_for_gps_within_groups
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_xls_groups_within_repeats
============================ 30 failed, 341 passed, 15 skipped, 47 warnings in 620.81s (0:10:20) =============================
```

https://github.com/kobotoolbox/kobocat/commit/76cfa0704795dce267187f81a424ba64bd7ee8b3 fixes:
```
FAILED onadata/apps/logger/tests/test_simple_submission.py::TestSimpleSubmission::test_corrupted_submission
FAILED onadata/apps/logger/tests/test_simple_submission.py::TestSimpleSubmission::test_simple_yes_submission
FAILED onadata/apps/logger/tests/test_simple_submission.py::TestSimpleSubmission::test_start_time_boolean_properly_set
FAILED onadata/apps/logger/tests/test_simple_submission.py::TestSimpleSubmission::test_start_time_submissions
```
Tests failed because `pyxform` now doesn't like questions having the same name as surveys. The survey before the changed one already has a question as a no spaces truncation of the survey title, so style is copied over

https://github.com/kobotoolbox/kobocat/commit/b3911123089afa4cd17ad5d6526e48ea801e8ab0 fixes:
```
FAILED onadata/apps/main/tests/test_csv_export.py::TestExport::test_csv_export_output
```
Test failed because the duplicate naming error message is changed as error is thrown for form-question names, as well as group names

https://github.com/kobotoolbox/kobocat/commit/fb3a9fd0fdfab73355a524a0650ae8b5c7716b92 and https://github.com/kobotoolbox/kobocat/commit/e394b1a587114f395cd9df2bf5427acf1d14c236 fixes:
```
FAILED onadata/apps/main/tests/test_csv_export.py::TestExport::test_csv_export_output
FAILED onadata/apps/main/tests/test_process.py::TestProcess::test_xls_export_content
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_build_sections_from_survey
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_delimiter_replacement_works_generated_multi_select_fields
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_generation_of_multi_selects_works
FAILED onadata/apps/viewer/tests/test_exports.py::TestExports::test_column_header_delimiter_export_option
FAILED onadata/apps/viewer/tests/test_exports.py::TestExports::test_csv_http_response
FAILED onadata/apps/viewer/tests/test_exports.py::TestExports::test_csv_without_na_values
FAILED onadata/apps/viewer/tests/test_exports.py::TestExports::test_split_select_multiple_export_option
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_csv_column_indices_in_groups_within_repeats
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_csv_columns_for_gps_within_groups
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_csv_dataframe_export_to
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_csv_export_with_df_size_limit
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_format_mongo_data_for_csv
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_split_select_multiples
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_xls_columns
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_xls_columns_for_gps_within_groups
FAILED onadata/apps/viewer/tests/test_pandas_mongo_bridge.py::TestPandasMongoBridge::test_xls_groups_within_repeats
```
Tests failed because https://github.com/XLSForm/pyxform/pull/410 changed bind types of `select*` questions from `select` and `select1` to `string`.  It's likely intention of `question.bind.get("type") == MULTIPLE_SELECT_BIND_TYPE` is to make sure the type is `select` and not `select1`. `select all that apply` should just replace the logic

https://github.com/kobotoolbox/kobocat/commit/be4eb8a19ab20e8dc5cb47de3e52874fc959ec1b fixes:
```
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_child_record_parent_table_is_updated_when_sheet_is_renamed
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_delimiter_replacement_works_for_generated_gps_fields
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_delimiter_replacement_works_generated_multi_select_fields
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_generation_of_gps_fields_works
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_generation_of_mongo_encoded_fields_works
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_generation_of_multi_selects_works
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_to_xls_export_generates_valid_sheet_names
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_to_xlsx_export_respects_custom_field_delimiter
FAILED onadata/apps/viewer/tests/test_export_builder.py::TestExportBuilder::test_to_xlsx_export_works
```
Tests failed because  [`pyxform.builder.create_survey_from_xls`](https://github.com/XLSForm/pyxform/commit/077972c5) now needs a parameter to specify the names of the surveys

### CODE REVIEW

File: onadata/apps/viewer/tests/test_export_builder.py [hunk 1]
```diff
@@ -677,7 +679,8 @@ def test_to_xls_export_generates_valid_sheet_names(self):
 
     def test_child_record_parent_table_is_updated_when_sheet_is_renamed(self):
         survey = create_survey_from_xls(_logger_fixture_path(
-            'childrens_survey_with_a_very_long_name.xls'))
+            'childrens_survey_with_a_very_long_name.xls'),
+            default_name='childrens_survey_with_a_very_long_name.xls')
```
Comment (joshuaberetta) [inline onadata/apps/viewer/tests/test_export_builder.py:683]: No really an issue, but `.xls` extension is included here in the `default_name`

File: onadata/apps/viewer/tests/test_export_builder.py [hunk 2]
```diff
@@ -759,7 +763,8 @@ def test_type_conversion(self):
 
     def test_xls_convert_dates_before_1900(self):
         survey = create_survey_from_xls(viewer_fixture_path(
-            'test_data_types/test_data_types.xls'))
+            'test_data_types/test_data_types.xls'),
+            default_name='test_data_types/test_data_types.xls')
```
Comment (joshuaberetta) [inline onadata/apps/viewer/tests/test_export_builder.py:767]: Same comment as above.

File: onadata/libs/constants.py [hunk 3]
```diff
@@ -35,4 +35,3 @@
 
 CAN_ADD_DATADICTIONARY = 'add_datadictionary'
 CAN_CHANGE_DATADICTIONARY = 'change_datadictionary'
-CAN_DELETE_DATADICTIONARY = 'delete_datadictionary'
```
Comment (joshuaberetta) [inline onadata/libs/constants.py:38]: What was the reason for removing `CAN_DELETE_DATADICTIONARY`?

**Comment by joshuaberetta** [comment]
Note to use `requirements.in` file and `pip-compile` rather than editing the `*.txt` files.

**Review by joshuaberetta — APPROVED** [review]

File: onadata/libs/constants.py (hunk 3 above)
Comment (duvld) [reply onadata/libs/constants.py:38]: Good catch :sweat_smile:

**Comment by jnm** [comment]
@noliveleger, we missed an [important caveat](https://stackoverflow.com/a/321941/1141214) on [`is_equal_xml()`](https://github.com/kobotoolbox/kobocat/blob/deca4ba3908d3838f142851babafc68336fcf10c/onadata/libs/tests/utils/xml.py#L36-L37)
> you have no other constructs than text and elements to worry about

As a result, pretty significant differences in XML, like the example below, were sailing through [unit tests](https://github.com/kobotoolbox/kobocat/blob/deca4ba3908d3838f142851babafc68336fcf10c/onadata/apps/api/tests/viewsets/test_xform_list_api.py#L256-L257) completely undetected. I've now removed `pyxform_version_agnostic()` and `is_equal_xml()` entirely (as part of #719), so it's no longer an issue.

```diff
@@ -1,9 +1,9 @@
-(Pdb) print(pyxform_version_agnostic(content))
+(Pdb) print(pyxform_version_agnostic(form_xml % data))
 <?xml version="1.0" encoding="utf-8"?>
 <h:html xmlns="http://www.w3.org/2002/xforms" xmlns:ev="http://www.w3.org/2001/xml-events" xmlns:h="http://www.w3.org/1999/xhtml" xmlns:jr="http://openrosa.org/javarosa" xmlns:odk="http://www.opendatakit.org/xforms" xmlns:orx="http://openrosa.org/xforms" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
   <h:head>
     <h:title>transportation_2011_07_25</h:title>
-    <model odk:xforms-version="1.0.0">
+    <model>
       <instance>
         <transportation_2011_07_25 id="transportation_2011_07_25">
           <formhub>
@@ -47,19 +47,19 @@
           </meta>
         </transportation_2011_07_25>
       </instance>
-      <bind nodeset="/transportation_2011_07_25/transport/available_transportation_types_to_referral_facility" type="string"/>
+      <bind nodeset="/transportation_2011_07_25/transport/available_transportation_types_to_referral_facility" type="select"/>
       <bind nodeset="/transportation_2011_07_25/transport/available_transportation_types_to_referral_facility_other" relevant="selected(../available_transportation_types_to_referral_facility, 'other')" type="string"/>
-      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/ambulance/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'ambulance')" type="string"/>
-      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/bicycle/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'bicycle')" type="string"/>
-      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/boat_canoe/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'boat_canoe')" type="string"/>
-      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/bus/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'bus')" type="string"/>
-      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/donkey_mule_cart/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'donkey_mule_cart')" type="string"/>
-      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/keke_pepe/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'keke_pepe')" type="string"/>
-      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/lorry/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'lorry')" type="string"/>
-      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/motorbike/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'motorbike')" type="string"/>
-      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/taxi/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'taxi')" type="string"/>
+      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/ambulance/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'ambulance')" type="select1"/>
+      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/bicycle/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'bicycle')" type="select1"/>
+      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/boat_canoe/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'boat_canoe')" type="select1"/>
+      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/bus/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'bus')" type="select1"/>
+      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/donkey_mule_cart/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'donkey_mule_cart')" type="select1"/>
+      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/keke_pepe/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'keke_pepe')" type="select1"/>
+      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/lorry/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'lorry')" type="select1"/>
+      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/motorbike/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'motorbike')" type="select1"/>
+      <bind nodeset="/transportation_2011_07_25/transport/loop_over_transport_types_frequency/taxi/frequency_to_referral_facility" relevant="selected( /transportation_2011_07_25/transport/available_transportation_types_to_referral_facility , 'taxi')" type="select1"/>
       <bind nodeset="/transportation_2011_07_25/meta/instanceID" readonly="true()" type="string"/>
-      <bind nodeset="/transportation_2011_07_25/formhub/uuid" type="string" calculate="'ad758136cb284523815120ba0df3bb78'"/>
+      <bind calculate="'ad758136cb284523815120ba0df3bb78'" nodeset="/transportation_2011_07_25/formhub/uuid" type="string"/>
     </model>
   </h:head>
   <h:body>
```

**Pull request closed without merging**